subsys/bluetooth/mesh/gatt_services.c

```diff
@@ -1017,46 +993,22 @@ int bt_mesh_proxy_prov_enable(void)
 	}
 
 
-	return 0;
+	return bt_gatt_service_register(&prov_svc);
 }
 
-int bt_mesh_proxy_prov_disable(bool disconnect)
+int bt_mesh_proxy_prov_disable(void)
 {
-	int i;
+	int ret;
 
 	BT_DBG("");
 
-	if (gatt_svc == MESH_GATT_NONE) {
-		return -EALREADY;
-	}
-
-	if (gatt_svc != MESH_GATT_PROV) {
-		return -EBUSY;
-	}
```

Comment:
The removal of these checks everywhere makes me a bit uncomfortable. Are they redundant because of checks in the gatt subsystem?